cou/utils/juju_utils.py

```diff
@@ -266,6 +313,38 @@ async def connect(self) -> None:
         )
 
     @retry
+    async def get_applications(self) -> dict[str, COUApplication]:
+        """Return list of applications with all relevant information.
+
+        :returns: list of application with all information
+        :rtype: list[COUApplication]
+        """
+        # note(rgildein): We get the applications from the Juju status, since we can get more
+        #                 information the status than from objects. e.g. workload_version for unit
+        full_status = await self.get_status()
+        machines = await self.get_model_machines()
+
+        return {
+            app: COUApplication(
+                name=app,
+                can_upgrade_to=status.can_upgrade_to,
+                charm=status.charm,
```

Comment:
This is not what we need, value like this look like `'ch:amd64/focal/ubuntu-24'` and it should be just `ubuntu`.